strings/base_meta.h

```diff
@@ -119,12 +119,12 @@ namespace winrt::impl
 
     template <typename T>
 #if defined(__clang__)
-#if __has_declspec_attribute(uuid)
+#if __has_declspec_attribute(uuid) && defined(WINRT_IMPL_IUNKNOWN_DEFINED)
     inline const guid guid_v{ __uuidof(T) };
```

Comment:
Can't fold. msvc doesn't support `__has_declspec_attribute()`. Would have to duplicate.

Reply:
Looks like v16.0 ("trunk" on godbolt) [doesn't treat it as `constexpr` yet](https://gcc.godbolt.org/z/c1cqKK818) so probably too soon. At least would have to put version guards around it, so we don't force people to upgrade clang.

Reply:
Moving at C++ speed - leave it for another 5 years and we'll see. 

![WaitingSkeletonGIF](https://user-images.githubusercontent.com/9845234/188978762-ea9c17f7-65cd-432c-9e08-7fc57274721a.gif)

Reply:
This might be caused by the fact that the version of clang on Godbolt runs on Linux. Windows builds of Clang support it fine (if constexpr was an issue clang would fail building, but here it fails linking because there is no main):
```
C:\Users\Sylveon\Downloads>type constexpr.cpp
typedef struct _GUID {
    unsigned long  Data1;
    unsigned short Data2;
    unsigned short Data3;
    unsigned char Data4[ 8 ];
    } GUID;

struct __declspec(uuid("ce2edeb3-10b7-41a0-a579-faab4fcaa73e")) S;

// uncomment this line to get an error
constexpr _GUID xx = __uuidof(S);

int major = __clang_major__ ;
int minor = __clang_minor__ ;
C:\Users\Sylveon\Downloads>clang-cl -v
clang version 14.0.5
Target: i686-pc-windows-msvc
Thread model: posix
InstalledDir: C:\Program Files\Microsoft Visual Studio\2022\Preview\VC\Tools\Llvm\bin

C:\Users\Sylveon\Downloads>clang-cl constexpr.cpp
LINK : fatal error LNK1561: entry point must be defined
clang-cl: error: linker command failed with exit code 1561 (use -v to see invocation)
```

Reply:
If we can enable Clang in the CI build we can more easily introduce such a change without all the guesswork.